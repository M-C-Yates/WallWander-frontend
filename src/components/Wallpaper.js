import React from 'react';

export const Wallpaper = (props) => {
	const { image, author, removeImage } = props;
	const urlArr = image.secureUrl.split('/');
	// urlArr[6] = 'f_auto,h_1440,w_2560,c_limit,q_auto:best';
	urlArr[6] = 'f_auto,c_limit';

	image.secureUrl = urlArr.join('/');
	return (
		<div className="wallpaper">
			<figure className="wallpaper--figure">
				<img
					className="wallpaper--img"
					src={image.secureUrl}
					alt={image.title}
				/>
				<figcaption className="wallpaper--info">
					<p>height: {image.height}px</p>
					<p>width: {image.width}px</p>
					<p>format: {image.format}</p>
					<p>Views: {image.views}</p>
					{author ? (
						<button className="btn btn-danger" onClick={removeImage}>
							Delete
						</button>
					) : (
						<React.Fragment />
					)}
				</figcaption>
			</figure>
		</div>
	);
};

export default Wallpaper;
