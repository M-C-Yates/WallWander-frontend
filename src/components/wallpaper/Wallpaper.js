import React from 'react';
import { Link } from '@reach/router';

const Favorite = React.lazy(() => import('./Favorite'));

export const Wallpaper = (props) => {
	const { wallpaperData, favoriteData, removeImage } = props;
	const { author, image, tags, user } = wallpaperData;

	const urlArr = image.secureUrl.split('/');
	urlArr[6] = 'f_auto,h_2560,w_1440,c_limit,q_auto:best';
	const optUrl = urlArr.join('/');

	return (
		<div className="wallpaper">
			<figure className="wallpaper--figure">
				<a href={image.secureUrl}>
					<img className="wallpaper--img" src={optUrl} alt={image.title} />
					<figcaption className="wallpaper--info">
						<p>height: {image.height}px</p>
						<p>width: {image.width}px</p>
						<p>format: {image.format}</p>
						<p>Views: {image.views}</p>
					</figcaption>
				</a>
			</figure>
			<p>
				uploader:{' '}
				<Link className="nav--link" to={`/profile/${image.userId}`}>
					{image.authorName}
				</Link>
			</p>

			<Favorite author={author} user={user} favoriteData={favoriteData} />

			{author ? (
				<button className="btn btn-danger" onClick={removeImage}>
					Delete
				</button>
			) : (
				<React.Fragment />
			)}

			<footer>
				icons made by <a href="www.flaticon.com">flaticon</a>
			</footer>
		</div>
	);
};

export default Wallpaper;
