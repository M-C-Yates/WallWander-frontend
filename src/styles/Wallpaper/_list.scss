.grid {
	display: block;
	max-width: 100%;
	width: 100vw;
	min-height: 100vh;
	height: 100%;
	margin: 0 auto;
	margin-bottom: 10%;

	div {
		display: inline-block;
	}
}

.thumb {
	width: 100%;
	height: auto;

	&--fig {
		width: 100%;
		height: 100%;
		clip-path: fill-box;
	}

	&--img {
		width: 100%;
		height: 100%;
		box-shadow: var(--boxshadow);

		border-radius: 5px;
		-webkit-border-radius: 5px;
		-moz-border-radius: 5px;
		-ms-border-radius: 5px;
		-o-border-radius: 5px;

		transition: all 0.4s ease;
		-webkit-transition: all 0.4s ease;
		-moz-transition: all 0.4s ease;
		-ms-transition: all 0.4s ease;
		-o-transition: all 0.4s ease;
	}

	&--info {
		display: flex;
		flex-direction: column;
		justify-content: space-around;
		align-items: center;

		position: absolute;
		width: 100%;
		height: 100%;
		left: 0%;
		top: 0%;

		color: var(--font-primary);
		opacity: 0;
		background-color: var(--background-primary-alpha);
		z-index: 2;

		border-radius: 5px;
		-webkit-border-radius: 5px;
		-moz-border-radius: 5px;
		-ms-border-radius: 5px;
		-o-border-radius: 5px;
		transition: all 0.4s ease;
		-webkit-transition: all 0.4s ease;
		-moz-transition: all 0.4s ease;
		-ms-transition: all 0.4s ease;
		-o-transition: all 0.4s ease;

		&--box {
			width: 100%;
			height: 100%;
			top: 0%;
			left: 0%;
		}
	}

	&:hover &--img {
		filter: blur(2px);
		-webkit-filter: blur(2px);
	}

	&:hover &--info {
		opacity: 1;
	}
}
