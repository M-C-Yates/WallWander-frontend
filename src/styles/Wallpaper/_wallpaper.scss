.wallpaper {
	width: 100vw;
	max-width: 100%;
	height: auto;

	display: flex;
	justify-content: space-between;
	width: 100vw;
	max-width: 100%;

	&-fit {
		position: relative;
		max-height: 100vw;
		width: 100vw;
		max-width: 90%;
	}

	&-figure {
		position: absolute;
		top: 0;
		bottom: 0;
		left: 0;
		right: 0;

		a {
			&:focus {
				border: none;
				outline: none;
			}
		}
	}
	&-img {
		position: absolute;
		top: 0;
		bottom: 0;
		left: 0;
		right: 0;

		display: inline-block;

		width: 100%;
		height: auto;
		max-width: 85vw;
		max-height: 85vh;
		margin: 0 auto;
		box-shadow: var(--boxshadow);

		border-radius: 5px;
		-webkit-border-radius: 5px;
		-moz-border-radius: 5px;
		-ms-border-radius: 5px;
		-o-border-radius: 5px;
		z-index: 0;
	}

	&-sidebar {
		display: flex;
		flex-direction: column;
		justify-content: space-between;
		align-items: center;

		width: 10%;
		height: 100rem;
		margin-left: 2rem;

		background-color: var(--background-primary);
		box-shadow: var(--boxshadow);

		border-radius: 5px;
		-webkit-border-radius: 5px;
		-moz-border-radius: 5px;
		-ms-border-radius: 5px;
		-o-border-radius: 5px;
	}

	&-container {
		display: flex;
		flex-direction: column;
		justify-content: space-between;
		align-items: flex-start;
		padding: 1rem;

		width: 100%;
		height: 40%;
	}

	&-uploader {
		display: flex;
		flex-direction: column;
		align-items: center;
	}

	&-info {
		p {
			padding: 0.25rem;
			margin: 0.5rem;
		}
	}

	&-actions {
		display: flex;
		justify-content: center;
		align-items: center;
	}
}

@include respond(tabLand) {
	.wallpaper {
		flex-direction: column;
		overflow-x: hidden;

		&-fit {
			max-height: 100%;
			position: static;
			width: 100vw;
			max-width: 100%;
			margin-bottom: 2rem;
		}

		&-figure {
			position: static;
			width: 95vw;
			max-width: 100%;
		}

		&-img {
			position: static;
			width: 95vw;
			max-width: 100%;
		}

		&-sidebar {
			flex-direction: column-reverse;
			align-self: center;
			width: 30vw;
			height: 100%;

			margin-bottom: 5rem;
		}
	}
}

@include respond(phone) {
	.wallpaper {
		&-fit {
			max-height: 100%;
			position: static;
			width: 100vw;
			max-width: 100%;
			margin-bottom: 2rem;
		}

		&-figure {
			position: static;
			width: 100vw;
			max-width: 100%;
		}

		&-img {
			position: static;
			width: 97vw;
			max-width: 100%;
		}

		&-sidebar {
			flex-direction: column-reverse;
			padding-top: 1rem;

			width: 100vw;
			max-width: 100%;
			margin: 0 auto;
		}
	}
}
