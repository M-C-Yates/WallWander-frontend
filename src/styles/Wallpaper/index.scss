@import './wallpaper';
@import './list';
@import './tags.scss';
