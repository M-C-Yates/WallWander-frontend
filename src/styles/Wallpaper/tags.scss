.tags {
	display: flex;
	flex-direction: column;
	width: 23rem;
	margin: 1rem auto;
	padding: 1rem;
	background-color: var(--background-secondary);

	box-shadow: var(--boxshadow);
	font-size: 1.6rem;
	border-radius: 5px;
	-webkit-border-radius: 5px;
	-moz-border-radius: 5px;
	-ms-border-radius: 5px;
	-o-border-radius: 5px;

	&-tagslist {
		display: flex;
		flex-grow: 1;
	}

	&-tag {
		display: inline-flex;
		flex-direction: row;
		justify-content: space-around;
		align-items: center;

		margin: 4px;
		padding-left: 5px;
		background-color: var(--background-primary);
		box-shadow: var(--boxshadow-sm);
		&-data {
			float: left;
			height: 100%;
		}

		&-nsfw {
			color: var(--fav-red);
		}
		border-radius: 5px;
		-webkit-border-radius: 5px;
		-moz-border-radius: 5px;
		-ms-border-radius: 5px;
		-o-border-radius: 5px;

		&:hover {
			background-color: var(--background-secondary);
			cursor: pointer;
		}
	}

	&-form {
		width: 100%;
		height: 100%;

		&-input {
			width: 10rem;
			height: 3rem;
			padding: 0.5rem 0.2rem;
			background-color: var(--background-primary);
			color: var(--font-primary);
			border-radius: 5px 0 0 5px;
			-webkit-border-radius: 5px 0 0 5px;
			-moz-border-radius: 5px 0 0 5px;
			-ms-border-radius: 5px 0 0 5px;
			-o-border-radius: 5px 0 0 5px;
		}

		&-select {
			width: 6rem;
			height: 100%;
			background-color: var(--background-primary);
			border: 1px solid transparent;
			color: var(--font-primary);
			outline: none;

			&:hover {
				cursor: pointer;
			}
		}

		&-option {
			background-color: var(--background-secondary);

			color: var(--font-primary);
			border: 1px solid transparent;
			outline: none;
		}
	}
}
