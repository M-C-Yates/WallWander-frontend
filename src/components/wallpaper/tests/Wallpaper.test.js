import React from 'react';
import { render, cleanup } from '@testing-library/react';

import Wallpaper from '../Wallpaper';

afterEach(cleanup);

jest.mock('react-redux', () => {
	return {
		useDispatch() {
			return { dispatch: jest.fn() };
		},
		useSelector() {
			return true;
		}
	};
});

describe('Wallpaper', () => {
	const wallpaperData = {
		id: '123abc',
		author: 'foobar',
		isAuthed: true,
		user: {
			id: '123abc',
			name: 'wooo',
			email: 'woo@bar.com'
		},
		image: {
			url: '1/11/1////1111111/1',
			secureUrl: '1///////1//1111111111111'
		}
	};
	it('should match snapshot', () => {
		const { asFragment } = render(<Wallpaper wallpaperData={wallpaperData} />);
		expect(asFragment).toMatchSnapshot();
	});
});
