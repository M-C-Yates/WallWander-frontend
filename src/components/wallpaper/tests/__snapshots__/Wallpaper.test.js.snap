// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wallpaper should match snapshot 1`] = `[Function]`;
